merkle/merkle_path.go

```diff
@@ -37,10 +41,10 @@ func CalcInclusionProofNodeAddresses(treeSize, index int64, maxBitLen int) ([]st
 			proof = append(proof, n)
 		} else if sibling == lastNodeAtLevel {
 			// The tree may skip levels because it's not completely filled in. These nodes
```

Comment:
I'm not sure that the distinction is that clear cut. The C++ is effectively retaining intermediate nodes with one child, which costs very little as there is only one version stored. Also the output of this calculation is in the correct form for our storage implementation. I'll try to add some text.